pkg/agent/attestor/node/node.go

```diff
@@ -201,14 +196,10 @@ func (a *attestor) newSVID(ctx context.Context, key *ecdsa.PrivateKey, bundle *b
 	defer cancel()
 
 	attestorName := "join_token"
-	var fetchStream nodeattestor.FetchAttestationData_Stream
+	var fetchStream nodeattestor.NodeAttestor_FetchAttestationDataClient
 	if a.c.JoinToken == "" {
-		plugins := a.c.Catalog.NodeAttestors()
-		if len(plugins) > 1 {
-			return nil, nil, errors.New("more than one node attestor configured")
-		}
-		attestor := plugins[0]
-		attestorName = attestor.Config().PluginName
+		attestor := a.c.Catalog.GetNodeAttestor()
```

Comment:
Nope. The contract with the Catalog is it will perform the proper constraints validation on the number and type of plugins. See pkg/agent/catalog, which guarantees that there is exactly one NodeAttestor. If a single plugin type is optional, that function will return a (plugin, bool) tuple to help force people to check whether or not the plugin is available.